src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerProgressReporter.cs

```diff
@@ -19,109 +16,69 @@ internal partial class SolutionCrawlerRegistrationService : ISolutionCrawlerRegi
         /// </summary>
         private class SolutionCrawlerProgressReporter : ISolutionCrawlerProgressReporter
         {
-            private readonly IAsynchronousOperationListener _listener;
+            private int _progressStartCount;
+            private int _progressEvaluateCount;
 
-            // use event map and event queue so that we can guarantee snapshot and sequential ordering of events from
-            // multiple consumer from possibly multiple threads
-            private readonly SimpleTaskQueue _eventQueue;
-            private readonly EventMap _eventMap;
+            public SolutionCrawlerProgressReporter()
+            {
+                _progressStartCount = 0;
+                _progressEvaluateCount = 0;
+            }
 
-            // this is to reduce number of times we report progress on same file. this is purely for perf
-            private string _lastReportedFilePath;
+            public event EventHandler<ProgressData> ProgressChanged;
 
-            private int _count;
+            public bool InProgress => _progressStartCount > 0;
 
-            public SolutionCrawlerProgressReporter(IAsynchronousOperationListener listener)
-            {
-                _listener = listener;
-                _eventQueue = new SimpleTaskQueue(TaskScheduler.Default);
-                _eventMap = new EventMap();
+            public void Start() => ChangeProgressStatus(ref _progressStartCount, ProgressStatus.Started);
+            public void Stop() => ChangeProgressStatus(ref _progressStartCount, ProgressStatus.Stoped);
+
+            private void Evaluate() => ChangeProgressStatus(ref _progressEvaluateCount, ProgressStatus.Evaluating);
+            private void Pause() => ChangeProgressStatus(ref _progressEvaluateCount, ProgressStatus.Paused);
 
-                _count = 0;
-            }
 
-            public bool InProgress
+            public void UpdatePendingItemCount(int pendingItemCount)
             {
-                get
+                if (_progressStartCount > 0)
                 {
-                    return _count > 0;
+                    var progressData = new ProgressData(ProgressStatus.PendingItemUpdated, pendingItemCount);
+                    OnProgressChanged(progressData);
                 }
             }
 
-            public event EventHandler<ProgressData> ProgressChanged
+            public IDisposable Evaluating()
```

Comment:
Can we rename this method.  It soundsl ike a question or state request.  Instead, it's a request to actual Evaluate.  It shoudl also have docs on the expectation of the caller.  i.e., presumably that the caller must dispose the object.

Reply:
sure. i can add doc comment.

Reply:
thanks :)